litestar/plugins.py

```diff
@@ -161,3 +159,41 @@ def to_openapi_schema(self, annotation: Any, schema_creator: SchemaCreator, dto_
     OpenAPISchemaPluginProtocol,
     CLIPluginProtocol,
 ]
+
+PluginT = TypeVar("PluginT", bound=PluginProtocol)
+
+
+class PluginRegistry:
```

Comment:
This can extend the [Mapping ABC](https://docs.python.org/3/library/collections.abc.html#collections.abc.Mapping) to get a few more standard mapping methods for free.

Reply:
I guess that could be done. I was thinking we leave the interface minimal for now to make it a bit more flexible for future use cases, and I didn't really see a use case for `items()`, `values()`, etc. as this is now.